lib/trino-arrow/src/main/java/io/trino/arrow/ArrowTypeConverter.java

```diff
@@ -0,0 +1,89 @@
+package io.trino.arrow;
+
+import io.trino.arrow.type.PicosecondTimeType;
+import io.trino.arrow.type.PicosecondTimestampType;
+import io.trino.arrow.type.TimeWithValueTimezoneType;
+import io.trino.arrow.type.TimestampWithValueTimezoneType;
+import io.trino.spi.type.*;
+import org.apache.arrow.vector.types.DateUnit;
+import org.apache.arrow.vector.types.FloatingPointPrecision;
+import org.apache.arrow.vector.types.TimeUnit;
+import org.apache.arrow.vector.types.pojo.ArrowType;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.FieldType;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public final class ArrowTypeConverter
+{
+    private ArrowTypeConverter() {}
+
+    public static Field toArrowField(String name, Type type)
+    {
+        return switch (type) {
+            case ArrayType t -> {
+                // The list field itself has a List type
+                // Its child field represents the list's data type
+                Field elementField = new Field("element", new FieldType(true, toArrowType(t.getElementType()), null), null);
+                yield new Field(name, new FieldType(true, new ArrowType.List(), null), List.of(elementField));
+            }
+            case MapType t -> {
+                Field keyField = toArrowField("key", t.getKeyType());
+                Field valueField = toArrowField("value", t.getValueType());
+                Field structField = new Field("entries",
+                        FieldType.notNullable(new ArrowType.Struct()),
+                        List.of(keyField, valueField));
+                yield new Field(name, FieldType.nullable(new ArrowType.Map(false)), List.of(structField));
+            }
+            case RowType t -> {
+                List<Field> children = t.getFields().stream()
+                        .map(field -> toArrowField(field.getName().get(), field.getType()))
+                        .collect(Collectors.toList());
+                yield new Field(name, FieldType.nullable(new ArrowType.Struct()), children);
+            }
+            default -> new Field(name, FieldType.nullable(toArrowType(type)), null);
+        };
+    }
+    private static ArrowType toArrowType(Type type)
+    {
+        return switch (type) {
```

Comment:
this switch covers the types in the spi package, but from running tests it seems that other types are sometimes produced. is there some way to know what types could be produced as output?